src/sparsematrix.jl

```diff
@@ -725,11 +749,13 @@ function Base.sizehint!(S::SparseMatrixCSC, n::Integer)
 end
 
 # converting between SparseMatrixCSC types
-SparseMatrixCSC(S::AbstractSparseMatrixCSC) = copy(S)
+
 AbstractMatrix{Tv}(A::AbstractSparseMatrixCSC) where {Tv} = SparseMatrixCSC{Tv}(A)
-SparseMatrixCSC{Tv}(S::AbstractSparseMatrixCSC{Tv}) where {Tv} = copy(S)
+SparseMatrixCSC{Tv}(S::AbstractSparseMatrixCSC) where {Tv} = SparseMatrixCSC{Tv}(S)
 SparseMatrixCSC{Tv}(S::AbstractSparseMatrixCSC) where {Tv} = SparseMatrixCSC{Tv,eltype(getcolptr(S))}(S)
-SparseMatrixCSC{Tv,Ti}(S::AbstractSparseMatrixCSC{Tv,Ti}) where {Tv,Ti} = copy(S)
```

Comment:
isn't this function needed?

Reply:
I am not so sure.  I even think it is a bug - if some subtype `T<:AbstractSparseMatrixCSC` implements `copy(m::T)::T`, then the we would have for `typeof(m)==T`:

`typeof(SparseMatrixCSC(m))==T`. 

Just verified this under 1.10...

Reply:
IMHO  `SparseMatrixCSC(::AbstractSparseMatrixCSC)` should always return a `SparseMatrixCSC` .

Reply:
That seems reasonable.

Reply:
If we consider the current implementation as a bug to be fixed, we might define

`SparseMatrixCSC(A::AbstractSparseMatrixCSC)=SparseMatrixCSC(size(A)..., getcolptr(A), rowvals(A), nonzeros(A))`, in particular  as `sparse` is not supposed to return a `SparseMatrixCSC`.

The alternative is to keep this undefined, and assume that in order to have this one should write code like [this](https://github.com/SciML/LinearSolve.jl/blob/270b56).

Reply:
Seems reasonable!